packages/design-system/src/components/contextMenu/components/button.js

```diff
@@ -107,6 +109,14 @@ function ButtonWithRef(
   const autoGeneratedId = useMemo(uuidv4, []);
   const elementId = id || autoGeneratedId;
 
+  // Only supply a type if isToggled is boolean, otherwise leave type as undefined
+  const type =
+    typeof isToggled === 'boolean'
+      ? isToggled
+        ? BUTTON_TYPES.SECONDARY
+        : BUTTON_TYPES.TERTIARY
+      : undefined;
```

Comment:
This is slightly copied logic from the ToggleButton, but it's not a direct copy, so I feel it's okay to include here. It could be generalized to somewhere, but on the other hand, it's so simple!

Reply:
cc @samwhale @BrittanyIRL for visibility.

Reply:
Note that as far as I can tell, the menu buttons in the context menu use the button from the design-system, though never one of the specific types of buttons (primary, secondary, etc), but rather rely on the default common styles only.

Reply:
This is a nit, but `isToggled` to me implies that the button is toggled "on" whereas `isToggle` might convey that the button is a toggle-style button.

Reply:
💭 Honestly I don't think this should live in the design system's `Button`. The `Button` api is pretty flexible, and I don't think the responsibility of solving this specific case is the `Button`'s. I think that the parent component should decide and pass in the type.

Ex: In the future if a new toggle button were needed where the types were different - this would need to be changed since it spreads to all uses of the base `Button`.

Keeping this outside of the button would keep the api lightweight and flexible. I think the wrapper would look something like:

```js
function ToggleButton({ isToggled, ...props }) {
    
    const type =
      typeof isToggled === 'boolean'
        ? isToggled
          ? BUTTON_TYPES.SECONDARY
          : BUTTON_TYPES.TERTIARY
        : undefined;

    return (
        <Button {...props} type={type} />
    )
}
```

---

However, I know that this is practically done and the suggestion above isn't necessary to make this work. So no pressure either way.

Reply:
> This is a nit, but `isToggled` to me implies that the button is toggled "on" whereas `isToggle` might convey that the button is a toggle-style button.

@BrookeGraham, that's also the intent of the code. If isToggled is set to true or false, it is a toggle button and it's toggled based on the truthiness. If isToggled is not set at all, it is not a toggle button at all. It's my way of decoding both the fact that this is a toggle button and that this toggle button is respectively toggled on or off into a single prop.

@samwhale, that's a good point. I moved it completely out of the component now and into a local button in the floating menu codebase in 3c9de899da.

Reply:
@samwhale, just re-read your comment. This code didn't actually live in the design-system's `Button` but in the design-system's context menu's `Button` so it was a bit more specific than that. Regardless, I have moved it out of the design-system entirely, as having toggl-able context menu buttons (for now) is specific to the floating design menu.

Reply:
Ah, I missed that! but great, I like the change 😄